addons/mail/wizard/mail_compose_message.py

```diff
@@ -507,21 +499,25 @@ def _process_state(self, mail_values_dict):
                 mail_values['failure_type'] = 'mail_bl'
                 # Do not post the mail into the recipient's chatter
                 mail_values['is_notification'] = False
-            elif optout_emails and mail_to in optout_emails:
-                mail_values['state'] = 'cancel'
-                mail_values['failure_type'] = 'mail_optout'
-            elif done_emails and mail_to in done_emails and not mailing_document_based:
-                mail_values['state'] = 'cancel'
-                mail_values['failure_type'] = 'mail_dup'
-            # void of falsy values -> error
-            elif not mail_to:
+            # void or falsy values -> error
+            elif not any(recipients['mail_to']):
                 mail_values['state'] = 'cancel'
                 mail_values['failure_type'] = 'mail_email_missing'
-            elif not mail_to_normalized:
+            elif not any(recipients['mail_to_normalized']):
                 mail_values['state'] = 'cancel'
                 mail_values['failure_type'] = 'mail_email_invalid'
+            elif optout_emails and all(
+                    mail_to in optout_emails for mail_to in recipients['mail_to']
```

Comment:
Note (improvement on existing code, not completely related to your dev :) ) : opt out emails seems to contain normalized emails (even if does not seem to be specified, but in 15 code uses normalize emails), so shouldn't we compare normalized emails ? Or to be sure, normalize emails in ``optout_emails``(if possible, otherwise keep input when not able) and try to compare normalized emails ?